src/main/java/gov/nasa/pds/tools/validate/TargetExaminer.java

```diff
@@ -221,4 +221,37 @@ public static ArrayList<String> getTargetContent(URL url, String nodeCheck, Stri
 
     return (fieldContent);
   }
+  public static boolean isTargetALabel(URL url) {
+    try {
+      InputSource source = Utility.getInputSourceByURL(url);
+      SAXSource saxSource = new SAXSource(source);
+      saxSource.setSystemId(url.toString());
+      DocumentInfo docInfo = LabelParser.parse(saxSource); // Parses a label.
+      @SuppressWarnings("unused")
+      List<TinyNodeImpl> xmlModels = new ArrayList<>();
+      XMLExtractor extractor = new XMLExtractor(docInfo);
+      xmlModels = extractor.getNodesFromDoc("logical_identifier");
+      return true;
+    } catch (Throwable t) {
```

Comment:
I'd prefer to see
```java
} catch (RuntimeException ex) {
  throw ex;
} catch (Exception ex) {
  return false;
}
```
otherwise serious environmental situations like `OutOfMemoryError` will be treated as the target not being a label when it should abort the program. (This is all due to the [defect in the way the Java exception hierarchy was defined from the beginning](https://github.com/NASA-PDS/nasa-pds.github.io/wiki/Java-Exceptions-Gotcha).)

Reply:
The errors that we have to catch are runtime errors unfortunately and not always predictable -- cannot enumerate them a priori. I agree that hiding out of memory errors is bad, but it is not likely that this would be the sole function that caused it since this is not the first function to read the file.  Nothing is done with the data that is read and all objects allocated here are disposed of immediately, well at the GC's leisure, making the risk of catching throwable here very minimal.

Reply:
Wait … so the functions `getInputSourceByURL`, `parse`, and/or `getNodesFromDoc` actually throw `RuntimeException`s on purpose instead of using some subclass of `Exception`?

Reply:
Yes plus extensions of exception and not sure why some are and are not extensions of runtime - they do not throw runtime but their extensions of runtime. That way you do not need to explicitly catch exceptions since we never use a URL that would cause one. Not my favorite libraries nor is catching exceptions my favorite way to test that something is valid - would rather have a library like xmllint say true/false on a comparison of the file and the schema. The schema is not fixed and part of the file being read so how do you read the file to get the schema to do the check, catch throwable since any exception can occur during the process. I would even take a JAXB mapping that condensed the exceptions down to a one or two rather but it requires generating code from schema that is not stationary.

If it would make you happier, could catch out of memory first, and raise again rather than null handler in this silly do nothing function. Could just make it exception instead of throwable and deal with the odd cases of runtime when they appear too.

Again, do not really care because the risk of this function masking some deep problem is really, really small. I do get the panic over violating best practices though.

Reply:
Nah … I mean … there are just _so many_ `RuntimeException`s … everything from `AnnotationTypeMismatchException` to `ConcurrentModificationException` to `IllegalStateException` to `SecurityException` to `SystemException`.

It's all low-level stuff and not the kind of thing you usually want to mix with application logic … I blame Steve Hughes for conflating the two kinds of exceptions in the first place 😆

Anyway, leave it catching `Throwable` and call this resolved.

Just my mind is still doing this: 🤯😂